src/Entity/Coalition/Cause.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+namespace App\Entity\Coalition;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use App\Entity\Adherent;
+use App\Entity\EntityIdentityTrait;
+use App\Entity\Event\CauseEvent;
+use App\Entity\ImageOwnerInterface;
+use App\Entity\ImageTrait;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Timestampable\Traits\TimestampableEntity;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\Serializer\Annotation as SymfonySerializer;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ApiResource(
+ *     attributes={
+ *         "pagination_client_items_per_page": true,
+ *         "order": {"name": "ASC"}
+ *     },
+ *     collectionOperations={
+ *         "get": {
+ *             "path": "/causes",
+ *             "normalization_context": {
+ *                 "groups": {"cause_read"}
+ *             },
+ *         },
+ *     },
+ *     itemOperations={
+ *         "get": {
+ *             "path": "/causes/{id}",
+ *             "normalization_context": {"groups": {"cause_read"}},
+ *             "requirements": {"id": "%pattern_uuid%"}
+ *         }
+ *     },
+ * )
+ *
+ * @ORM\Table(
+ *     uniqueConstraints={
+ *         @ORM\UniqueConstraint(name="cause_uuid_unique", columns="uuid"),
+ *         @ORM\UniqueConstraint(name="cause_name_unique", columns="name")
+ *     }
+ * )
+ * @ORM\Entity
+ */
+class Cause implements ImageOwnerInterface
+{
+    use EntityIdentityTrait;
+    use TimestampableEntity;
+    use ImageTrait;
+
+    /**
+     * @var UploadedFile|null
+     *
+     * @Assert\Image(
+     *     maxSize="5M",
+     *     mimeTypes={"image/jpeg", "image/png"}
+     * )
+     */
+    protected $image;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column
+     *
+     * @SymfonySerializer\Groups({"cause_read"})
+     */
+    private $name;
```

Comment:
should we use `EntityNameSlugTrait` to be able to use `slug` field for the URL?

Reply:
we will handle slugs in another PR